NewDawn/NewDawn/ViewControllers/SettingPageViewControllers/SettingPageViewController.swift

```diff
@@ -64,13 +64,23 @@ class SettingPageViewController: UIViewController {
                     let user_hometown = user_profile!.hometown
                     self.NameAgeText.text = user_firstname + ", " + String(user_age)
                     self.HomeTownText.text = user_hometown
-                    if user_profile!.mainImages.isEmpty == false {
-                        self.profileImage.downloaded(from:
-                            self.profileImage.getURL(path: user_profile!.mainImages[0].image_url))
+                    if let images_count = LocalStorageUtil.localReadKeyValue(key: "ImagesCount"){
+                        if user_profile!.mainImages.count == images_count as! Int {
+                            EditProfileTabelViewController.downloadOverwriteLocalImages(profile: user_profile!)
+                            EditProfileTabelViewController.downloadOverwriteLocalInfo(profile: user_profile!)
+                            self.profileImage.downloaded(from:
+                                self.profileImage.getURL(path: user_profile!.mainImages[0].image_url))
+                        }else{
+                            self.viewDidLoad()
```

Comment:
What happens if one image failed to upload? Would it infinitely try fetching images? If that’s the case, we should set a max_attempt and stop refreshing the page after that.

Reply:
I have set a maximum of 5 times to try download.